src/plugins/discover/public/application/main/components/layout/discover_documents.tsx

```diff
@@ -263,6 +264,12 @@ function DiscoverDocumentsComponent({
     useContextualGridCustomisations() || {};
   const additionalFieldGroups = useAdditionalFieldGroups();
 
+  const getCellRenderersAccessor = useProfileAccessor('getCellRenderers');
+  const cellRenderers = useMemo(() => {
+    const getCellRenderers = getCellRenderersAccessor(() => customCellRenderer ?? {});
+    return getCellRenderers();
```

Comment:
Unified Data Table cell renders are column based currently, so they don't work on individual records and can't be supported by `DocumentProfile` right now. I'm not sure if we need cell renderers on a per-record basis yet, but maybe there are some good use cases for it.

If we decide we need it, I'd recommend we add support for them and do the necessary Unified Data Table refactoring in a followup.

Reply:
Ah, sorry, I didn't know this was an intentional omission. Could this indicate that narrowing the composable profile interface by context type might be good? Because I would also have made that mistake when trying to implement a custom cell renderer.